### PR TITLE
feat: serde support for a basic structs

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
A feature request was to have serde support for some basic structs such as a PathQuery and an Element

## What was done?
Added Serde support as a feature

## Breaking Changes
No breaking changes.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced serialization and deserialization support for several structures and enums using the Serde library, enhancing data interchange capabilities.
	- Added a new feature flag `serde-support` to enable serialization features selectively.
	- Enhanced the `Query` structure to include a new field for conditional subqueries.

- **Bug Fixes**
	- Corrected authorship details in the package configuration.

- **Documentation**
	- Updated metadata in package configuration files to reflect new dependencies and features.

### CODE REVIEW

**Comment by QuantumExplorer** [comment]
@coderabbitai review

File: grovedb/Cargo.toml [hunk 1]
```diff
@@ -50,8 +51,9 @@ name = "insertion_benchmark"
 harness = false
 
 [features]
-default = ["full"]
+default = ["full", "serde-support"]
 proof_debug = ["grovedb-merk/proof_debug"]
+serde-support = ["serde", "grovedb-merk/serde-support", "indexmap/serde"]
```
Comment (lklimek) [inline grovedb/Cargo.toml:56]: I think `serde` is a de facto standard for the feature name

```suggestion
serde = ["dep:serde", "grovedb-merk/serde-support", "indexmap/serde"]
```

File: merk/Cargo.toml [hunk 2]
```diff
@@ -56,6 +57,7 @@ optional = true
 [features]
 default = ["full"]
 proof_debug = []
+serde-support = ["serde", "indexmap/serde"]
```
Comment (lklimek) [inline merk/Cargo.toml:60]:
```suggestion
serde = ["dep:serde", "indexmap/serde"]
```

File: grovedb/Cargo.toml (hunk 1 above)
Comment (QuantumExplorer) [reply grovedb/Cargo.toml]: doen

File: merk/src/proofs/query/query_item/mod.rs [hunk 3]
```diff
@@ -36,6 +40,149 @@
...
+            QueryItem::RangeTo(range_to) => {
+                serializer.serialize_newtype_variant("QueryItem", 5, "RangeTo", range_to)
+            }
+            QueryItem::RangeToInclusive(range_to_inclusive) => serializer
+                .serialize_newtype_variant(
+                    "QueryItem",
+                    6,
+                    "RangeToInclusive",
+                    &range_to_inclusive.end,
+                ),
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply merk/src/proofs/query/query_item/mod.rs:72]: no, range_to_inclusive does not support serde

**Review by lklimek — APPROVED** [review]

**Pull request merged**